packages/admin-api-client/src/admin-api-client.ts

```diff
@@ -88,17 +89,28 @@ export function createAdminApiClient({
   const getHeaders = generateGetHeaders(config);
   const getApiUrl = generateGetApiUrl(config, apiUrlFormatter);
 
-  const getGQLClientParams = generateGetGQLClientParams({
+  const getGQLClientParams = generateGetGQLClientParams<AdminOperations>({
     getHeaders,
     getApiUrl,
   });
 
-  const fetch = (...props: ApiClientRequestParams) => {
-    return graphqlClient.fetch(...getGQLClientParams(...props));
+  const fetch = <Operation extends keyof AdminOperations>(
```

Comment:
Are we able to set the `fetch` and `request` to types `ApiClientFetch` and `ApiClientRequest` respectively?

eg:
> const fetch: ApiClientFetch = <Operation extends keyof AdminOperations>(...) => {...}

Reply:
Actually, now that I'm looking at more carefully, I think we'd need to update the `ApiClientFetch` so that we can pass in `Operations` generic type for it to work - in this case `AdminOperations` needs to be passed in too right?

Reply:
Yeah, I believe this will do the trick:

```ts
  const fetch: ApiClientFetch<AdminOperations> = (
    ...props: Parameters<ApiClientFetch<AdminOperations>>
  ) => {
    return graphqlClient.fetch(...getGQLClientParams(...props));
  };
```

But maybe I should reintroduce the `Params` type I removed to make this clearer.

Reply:
Yeah, that would be cleaner 🤔